corehq/apps/users/views/mobile/users.py

```diff
@@ -1481,3 +1501,42 @@ def post(self, request, *args, **kwargs):
 
         # todo: process form data and activate the account
         return self.get(request, *args, **kwargs)
+
+@location_safe
+class CommCareUserConfirmAccountBySMSView(CommCareUserConfirmAccountView):
+    urlname = "commcare_user_confirm_account_sms"
+    default_expiry_duration_in_hours = 24
```

Comment:
Having a default value for expiry in the form would also let you get rid of this variable, which would be good.

Reply:
If the model defaults to 24 and the UI throws a validation error if the user clears it, you'd be guaranteed to have a value in `Domain.confirmation_link_expiry_time`.

Reply:
So along with form, model should also have a default then?

Reply:
The form shouldn't need a default if the model has one (the model can also not allow nulls), it just needs to validate.

Reply:
Ignore the comment about allowing nulls, I was thinking in SQL terms, forgot this is a couch model. I think you can
* Set a default value on the model attribute
* Remove the default value from the form
* Add validation to the form so that it errors if the user clears the expiry value

Reply:
I am little confused here. Based on above, when form loads, should we read existing value from the model(with a default set) and populate it? Along with it, make expiry field mandatory, such that if user intentionally removes the value then form validation will kick in to disallow submission?

Reply:
yes, exactly that

Reply:
Made the changes.